src/main.cpp

```diff
@@ -10,43 +11,61 @@ int main() {
 #endif
 	std::cout << "Hello, GC user!" << std::endl;
 	Heap heap;
-	std::cout << heap.ToString() << std::endl;
+	//heap.dump();
 
 	// TD based on "Garbage Collection" slide-set page 22
 	// Create TypeDescriptor for a class (Block x; Block y; int getDataPart; Block z;)
 	std::cout << std::endl;
 	std::cout << "Creating TypeDescriptor for class 'SlideBlock'…" << std::endl;
-	TypeDescriptor td = TypeDescriptor(16, new int[3]{0, 4, 12}, 3);
+	TypeDescriptor td = TypeDescriptor(32,
+									   new int[3]{0, 8, 24}, // For offset explanation see SlideBlock.hpp
+									   3);
 	std::cout << td.ToString() << std::endl;
 	if (heap.registerType("SlideBlock", td)) {
-		std::cout << "Successfully registered typeDescriptor!" << std::endl;
+		std::cout << "Successfully registered rawTypeDescriptor!" << std::endl;
 	} else {
-		std::cout << "Failed to register typeDescriptor!" << std::endl;
+		std::cout << "Failed to register rawTypeDescriptor!" << std::endl;
 		return 1;
 	}
 
 	std::cout << std::endl;
 	std::cout << "Allocating 'SlideBlock'…" << std::endl;
-	Block* blockFromTD = heap.alloc("SlideBlock");
-	std::cout << blockFromTD->ToString() << std::endl;
-	std::cout << heap.ToString() << std::endl;
+	UsedBlock* b1 = heap.alloc("SlideBlock");
+	// I just use "new" here to trigger the constructor
+	// FIXME: Something is wrong when the SlideBlock constructor is called and sets the pointers to null.
```

Comment:
You mean the pointers inside slideblock?

Reply:
Just ignore the `FIXME`. They were my helping comments and I forgot to remove them